libexec/rtld-elf/rtld_lock.c

```diff
@@ -427,16 +427,19 @@ _rtld_thread_init(struct RtldLockInfo *pli)
 		}
 #ifdef CHERI_LIB_C18N
 		tmplockinfo = *pli;
-#define WRAP(_target, _valid, _reg_args, _mem_args, _ret_args)	\
-	_target = tramp_intern(NULL, RTLD_COMPART_ID, &(struct tramp_data) {	\
-		.target = _target,						\
-		.defobj = obj,							\
-		.sig = (struct func_sig) {					\
-			.valid = _valid,					\
-			.reg_args = _reg_args, .mem_args = _mem_args,		\
-			.ret_args = _ret_args					\
-		}								\
-	})
+#define WRAP(_target, _valid, _reg_args, _mem_args, _ret_args)			\
```

Comment:
Perhaps avoid the unusual `do if` syntax by following a pattern I've seen several times before:

```
#define FOO(...) do {    \
......
} while (0)
```